src/Nethermind.Verkle/Fields/FpEElement/Element.Sqrt.cs

```diff
@@ -0,0 +1,316 @@
+        MultiplyMod(in acc, in z25, out acc);
+        SquareEqNTimes(acc, out acc, 5);
+        MultiplyMod(in acc, in z25, out acc);
+        SquareEqNTimes(acc, out acc, 5);
+        MultiplyMod(in acc, in z27, out acc);
+        SquareEqNTimes(acc, out acc, 8);
+        MultiplyMod(in acc, in z, out acc);
+        SquareEqNTimes(acc, out acc, 8);
+        MultiplyMod(in acc, in z, out acc);
+        SquareEqNTimes(acc, out acc, 6);
+        MultiplyMod(in acc, in z13, out acc);
+        SquareEqNTimes(acc, out acc, 7);
+        MultiplyMod(in acc, in z7, out acc);
+        SquareEqNTimes(acc, out acc, 3);
+        MultiplyMod(in acc, in z3, out acc);
+        SquareEqNTimes(acc, out acc, 13);
+        MultiplyMod(in acc, in z21, out acc);
+        SquareEqNTimes(acc, out acc, 5);
+        MultiplyMod(in acc, in z9, out acc);
+        SquareEqNTimes(acc, out acc, 5);
+        MultiplyMod(in acc, in z27, out acc);
+        SquareEqNTimes(acc, out acc, 5);
+        MultiplyMod(in acc, in z27, out acc);
+        SquareEqNTimes(acc, out acc, 5);
+        MultiplyMod(in acc, in z9, out acc);
+        SquareEqNTimes(acc, out acc, 10);
+        MultiplyMod(in acc, in z, out acc);
+        SquareEqNTimes(acc, out acc, 7);
+        MultiplyMod(in acc, in z255, out acc);
+        SquareEqNTimes(acc, out acc, 8);
+        MultiplyMod(in acc, in z255, out acc);
+        SquareEqNTimes(acc, out acc, 6);
+        MultiplyMod(in acc, in z11, out acc);
+        SquareEqNTimes(acc, out acc, 9);
+        MultiplyMod(in acc, in z255, out acc);
+        SquareEqNTimes(acc, out acc, 2);
+        MultiplyMod(in acc, in z, out acc);
+        SquareEqNTimes(acc, out acc, 7);
+        MultiplyMod(in acc, in z255, out acc);
+        SquareEqNTimes(acc, out acc, 8);
+        MultiplyMod(in acc, in z255, out acc);
+        SquareEqNTimes(acc, out acc, 8);
+        MultiplyMod(in acc, in z255, out acc);
+        SquareEqNTimes(acc, out acc, 8);
+        MultiplyMod(in acc, in z255, out acc);
+//acc = n^(Q-1)/2
+        MultiplyMod(in acc, in acc, out rootOfUnity);
+        MultiplyMod(in rootOfUnity, in z, out rootOfUnity);//n^Q
+        MultiplyMod(in acc, in z, out squareRootCandidate);//n^(Q+1)/2
+
+        return acc;
+    }
+
+//Given n, whose square root needs to be found, this method returns n^(Q*2^24), n^(Q*2^16), n^(Q*2^8), n^(Q) where p-1 = Q*2^s where s is 32 for this curve. This method takes n^Q as argument.
+    public static FE[] powersOfNq(in FE n){
+        FE[] arr = new FE[4];
+        FE res = n;
+        arr[0] = res;
+
+        for(int i=0;i<24;i++){
+            FE.MultiplyMod(res, res, out res);
+            if(i==7){
+                arr[1] = res;
+            }else if(i==15){
+                arr[2] = res;
+            }
+        }
+        arr[3] = res;
+        return arr;
+    }
+
+//Given a number x, this method returns y0, y1, y2, y3: where x = (2^24)*y0 + (2^16)*y1 + (2^8)*y2 + (2^0)*y3
+    public static ulong[] decomposeNumber(ulong x){
```

Comment:
use byte[] to represent the decomposed number

Reply:
public static byte[] decomposeNumber(ulong x){
    byte y0 = (byte)((x >> 24) & 0xFF); // Shift right by 24 bits and mask out the last 8 bits, then cast to byte
    byte y1 = (byte)((x >> 16) & 0xFF); // Shift right by 16 bits and mask out the last 8 bits, then cast to byte
    byte y2 = (byte)((x >> 8) & 0xFF);  // Shift right by 8 bits and mask out the last 8 bits, then cast to byte
    byte y3 = (byte)(x & 0xFF);         // Mask out the last 8 bits and cast to byte
    byte[] arr = {y0, y1, y2, y3}; 
    return arr;
}

Is this what you mean?